src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -266,13 +266,13 @@ private void ClearCurrentBinding()
             this.VisualStateManager.ClearBoundProject();
         }
 
-        private void ApplyBindingInformation(LegacyBindingConfiguration bindingConfig)
+        private void ApplyBindingInformation(BindingConfiguration bindingConfig)
         {
             SharedBindingConfig = null;
 
             // Set the project key that should become bound once the connection workflow has completed
-            this.VisualStateManager.BoundProjectKey = bindingConfig.Project.ProjectKey;
-            this.VisualStateManager.BoundProjectName = bindingConfig.Project.ProjectName;
+            this.VisualStateManager.BoundProjectKey = bindingConfig.Project.ServerProjectKey;
+            this.VisualStateManager.BoundProjectName = bindingConfig.Project.ServerProjectKey;
```

Comment:
We don't have the name in the new model and this class will be deleted after this sprint, so I just used the project key here instead